tests/ui/traits/trait-upcasting/type-checking-test-1.rs

```diff
@@ -15,7 +18,7 @@ fn test_specific(x: &dyn Foo) {
 fn test_unknown_version(x: &dyn Foo) {
     let _ = x as &dyn Bar<_>; // Ambiguous
                               //~^ ERROR non-primitive cast
-                              //~^^ ERROR the trait bound `&dyn Foo: Bar<_>` is not satisfied
+                              //[current]~^^ ERROR the trait bound `&dyn Foo: Bar<_>` is not satisfied
```

Comment:
I actually have no idea where this error message is coming from. It also seems just wrong? This is the error message I'd expect to get if we had `&x as &dyn Bar<_>`, not `x as &dyn Bar<_>`...